From 00c45306def1e373160967df6ebaafb9d1980d5b Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Mon, 9 Mar 2015 18:05:19 +0200
Subject: [PATCH 05/66] aarch64: defconfig: Add support for adding user files
 to output file-system

	Add an option to add user binaries / directories to the resulting
	filesystem.
	All files / directoeis placed under user_files/marvell will be
	copied to the "/marvell" directory, under the target root
	filesystem.

Change-Id: Ie289f7252939fcd8cfb4f3c5cfdac5dab49cc117
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/17384
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 configs/mvebu_armv8_le_defconfig      | 1 +
 configs/mvebu_armv8_tiny_le_defconfig | 1 +
 user_files/marvell/help.sh            | 5 +++++
 3 files changed, 7 insertions(+)
 create mode 100755 user_files/marvell/help.sh

diff --git a/configs/mvebu_armv8_le_defconfig b/configs/mvebu_armv8_le_defconfig
index ac51958..49fd82f 100644
--- a/configs/mvebu_armv8_le_defconfig
+++ b/configs/mvebu_armv8_le_defconfig
@@ -1,6 +1,7 @@
 BR2_aarch64=y
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TARGET_GENERIC_GETTY_PORT="ttyS0"
+BR2_ROOTFS_OVERLAY="user_files/"
 BR2_PACKAGE_BZIP2=y
 BR2_PACKAGE_DHRYSTONE=y
 BR2_PACKAGE_LMBENCH=y
diff --git a/configs/mvebu_armv8_tiny_le_defconfig b/configs/mvebu_armv8_tiny_le_defconfig
index be9ef1a..90312be 100644
--- a/configs/mvebu_armv8_tiny_le_defconfig
+++ b/configs/mvebu_armv8_tiny_le_defconfig
@@ -1,6 +1,7 @@
 BR2_aarch64=y
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TARGET_GENERIC_GETTY_PORT="ttyS0"
+BR2_ROOTFS_OVERLAY="user_files"
 BR2_TARGET_ROOTFS_CPIO=y
 BR2_TARGET_ROOTFS_TAR_GZIP=y
 BR2_PACKAGE_HOST_MKE2IMG=y
diff --git a/user_files/marvell/help.sh b/user_files/marvell/help.sh
new file mode 100755
index 0000000..1cceec2
--- /dev/null
+++ b/user_files/marvell/help.sh
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+echo "Files and directories added under this directory,"\
+     "will be placed under the \"/marvell\" directory"\
+     "in the target root filesystem."
-- 
1.9.1

